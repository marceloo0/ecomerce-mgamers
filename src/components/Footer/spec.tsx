import { screen } from '@testing-library/react'
import { renderWithTheme } from '../../utils/tests/helpers'

import { Footer } from '.'

jest.mock('../../hooks/useAuth', () => {
  return {
    useAuth() {
      return {
        user: {
          id: '1234',
          name: 'marcelo',
          avatar: '',
          email: 'marcelo@marcelo.com'
        }
      }
    }
  }
})

describe('<Footer />', () => {
  it('should render default', () => {
    const { container } = renderWithTheme(<Footer />)

    expect(screen.getByLabelText(/finaliza/i)).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })
})
