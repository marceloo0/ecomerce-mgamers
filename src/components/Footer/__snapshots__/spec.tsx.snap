// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer /> should render default 1`] = `
.c3 {
  width: 2rem;
  height: 2rem;
  border: 3px solid #7FFA45;
  border-radius: 50%;
  background-color: #201C2B;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c4 {
  text-transform: uppercase;
  color: #C7ECEC;
  font-size: 1.2rem;
  font-weight: 600;
}

.c10 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  font-family: Poppins,-apple-system,BlinkMacSystemFont,'Segoe UI',Roboto,Oxygen,Ubuntu,Cantarell,'Open Sans','Helvetica Neue',sans-serif;
  border: 1px solid transparent;
  border-radius: 0.4rem;
  cursor: pointer;
  color: #C7ECEC;
  background-color: #201C2B;
  padding: 1rem 3.2rem;
  font-size: 1.4rem;
}

.c10:hover {
  background-color: #332756;
}

.c6 {
  position: relative;
  width: 2.4rem;
  height: 2.4rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  background-color: transparent;
  border: none;
}

.c6 > svg {
  color: #7FFA45;
}

.c7 {
  width: 1.5rem;
  height: 1.5rem;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 0 0.5rem;
  bottom: 0;
}

.c0 {
  width: 100%;
  height: 100px;
}

.c1 {
  width: 100%;
  height: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  background-color: #2E2F42;
  padding: 12px;
}

.c9 {
  font-size: 1.8rem;
  color: #7FFA45;
  font-weight: 600;
  margin-left: 12px;
}

.c8 {
  font-size: 1.6rem;
  color: #415151;
  font-weight: 600;
}

.c5 {
  width: 0.1rem;
  height: 2rem;
  background-color: #C7ECEC;
  margin: 0 8px;
}

@media screen and (min-width:768px) {
  .c3 {
    width: 2.2rem;
    height: 2.2rem;
  }
}

@media screen and (min-width:768px) {
  .c4 {
    font-size: 1.6rem;
  }
}

@media screen and (min-width:768px) {
  .c6 {
    width: 3rem;
    height: 3rem;
  }
}

@media screen and (min-width:768px) {
  .c7 {
    width: 2.3rem;
    height: 2.3rem;
  }
}

@media (min-width:768px) {
  .c2 {
    display: none;
  }
}

@media (min-width:768px) {
  .c1 {
    -webkit-box-pack: end;
    -webkit-justify-content: flex-end;
    -ms-flex-pack: end;
    justify-content: flex-end;
  }
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    <div
      class="c2"
    >
      <div
        class="c3"
      >
        <span
          class="c4"
        >
          m
        </span>
      </div>
      <div
        class="c5"
      />
      <button
        class="c6"
      >
        <svg
          aria-label="shopify"
          class="c7"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 448 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M388.32,104.1a4.66,4.66,0,0,0-4.4-4c-2,0-37.23-.8-37.23-.8s-21.61-20.82-29.62-28.83V503.2L442.76,472S388.72,106.5,388.32,104.1ZM288.65,70.47a116.67,116.67,0,0,0-7.21-17.61C271,32.85,255.42,22,237,22a15,15,0,0,0-4,.4c-.4-.8-1.2-1.2-1.6-2C223.4,11.63,213,7.63,200.58,8c-24,.8-48,18-67.25,48.83-13.61,21.62-24,48.84-26.82,70.06-27.62,8.4-46.83,14.41-47.23,14.81-14,4.4-14.41,4.8-16,18-1.2,10-38,291.82-38,291.82L307.86,504V65.67a41.66,41.66,0,0,0-4.4.4S297.86,67.67,288.65,70.47ZM233.41,87.69c-16,4.8-33.63,10.4-50.84,15.61,4.8-18.82,14.41-37.63,25.62-50,4.4-4.4,10.41-9.61,17.21-12.81C232.21,54.86,233.81,74.48,233.41,87.69ZM200.58,24.44A27.49,27.49,0,0,1,215,28c-6.4,3.2-12.81,8.41-18.81,14.41-15.21,16.42-26.82,42-31.62,66.45-14.42,4.41-28.83,8.81-42,12.81C131.33,83.28,163.75,25.24,200.58,24.44ZM154.15,244.61c1.6,25.61,69.25,31.22,73.25,91.66,2.8,47.64-25.22,80.06-65.65,82.47-48.83,3.2-75.65-25.62-75.65-25.62l10.4-44s26.82,20.42,48.44,18.82c14-.8,19.22-12.41,18.81-20.42-2-33.62-57.24-31.62-60.84-86.86-3.2-46.44,27.22-93.27,94.47-97.68,26-1.6,39.23,4.81,39.23,4.81L221.4,225.39s-17.21-8-37.63-6.4C154.15,221,153.75,239.8,154.15,244.61ZM249.42,82.88c0-12-1.6-29.22-7.21-43.63,18.42,3.6,27.22,24,31.23,36.43Q262.63,78.68,249.42,82.88Z"
          />
        </svg>
      </button>
    </div>
    <span
      class="c8"
    >
      Valor total:
    </span>
    <span
      class="c9"
    >
      R$Â NaN
    </span>
    <div
      style="padding: 24px;"
    >
      <button
        aria-label="finaliza"
        class="c10"
      >
        <span>
          Finalizar compra
        </span>
      </button>
    </div>
  </div>
</div>
`;
